src/pages/lessons/[slug].tsx

```diff
@@ -432,41 +441,79 @@ const Lesson: React.FC<LessonProps> = ({initialLesson}) => {
               send(`RECOMMEND`)
             }
           })
-
         break
     }
-  }, [currentPlayerState, session_id])
+  }, [currentLessonState, session_id])
 
   React.useEffect(() => {
-    // when this effect is triggered, put the machine in the loading state (via
-    // `LOAD`). That state should take care of the rest with an Invoked
-    // Callback.
-    send({
-      type: 'LOAD',
-      lesson: initialLesson,
-      viewer,
+    // Keep lesson machine state in sync with
+    // videoService to control overlays and stuff
+    if (!isWaiting) {
+      hasEnded && send('COMPLETE')
+      isPaused ? send('PAUSE') : send('PLAY')
+    }
+  }, [hasEnded, isPaused, isWaiting])
+
+  React.useEffect(() => {
+    // Load the video resource
+    send('LOAD')
+    videoService.send({
+      type: 'LOAD_RESOURCE',
+      resource: initialLesson as any,
     })
   }, [initialLesson.slug])
```

Comment:
Feed the player machine with `initialLesson` each time we navigate to new lesson. Player machine then takes care of the rest.